Roman_Gvozdov/js/scripts.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Created by John Doe on 17.10.2017.
+ */
+document.onreadystatechange = function () {
+  if (document.readyState === 'interactive') {
+    initCalculator();
+  }
+};
+
+function initCalculator() {
+  var
+    calculator = document.getElementById('calculator'),
+    first = document.getElementById('first-number'),
+    second = document.getElementById('second-number'),
+    action = document.getElementById('action');
+
+  calculator.onsubmit = function(event) {
+	 
+    var
+      firstNumber = parseFloat(first.value),
+      secondNumber = parseFloat(second.value),
+      actionType = action.value.trim(),
+      result = calculate(firstNumber, secondNumber, actionType);
+    event.preventDefault();
+	while(typeof(result)!='Nan' || typeof(result)!='undefined'){
```

Comment:
Why do You use `while` loop here?
Use `if` condition instead.
Operator `typeof` can return only `number`, `string`, `boolean`, `undefined` or `object`.
And You can't detect `NaN` with this operator, because of `typeof(NaN) === "number"`.
Use `isNaN` function instead:
```js
var result = 'string' / 10;

console.log(isNaN(result)); // true
```